purchase_discount/tests/test_purchase_discount.py

```diff
@@ -126,3 +141,17 @@ def test_invoice(self):
             lambda x: x.purchase_line_id == self.po_line_3
         )
         self.assertEqual(line.discount, 0)
+
+    def test_invoice_subtotals(self):
+        invoice = self.env['account.invoice'].new({
+            'partner_id': self.env.ref('base.res_partner_3').id,
+            'purchase_id': self.purchase_order_check_amount.id,
+        })
+        invoice.purchase_order_change()
+        line = invoice.invoice_line_ids.filtered(
+            lambda x: x.purchase_line_id == self.po_line_check_amount
+        )
+        line.quantity = self.po_line_check_amount.product_qty
+        self.assertEqual(line.discount, 1)
+        self.assertAlmostEqual(line.price_subtotal,
+                               self.po_line_check_amount.price_subtotal)
```

Comment:
You must use fixed data in tests (for example, `15.03`), not references to other fields.